uavf_2024/imaging/target_tracker.py

```diff
@@ -33,8 +33,31 @@ def closest_match(self, target_desc: TargetDescription) -> Target3D:
         '''
         best_match, best_score = self._targets[0], 0
 
+        
+        '''Converts the target description value from an int into a numpy prob distribution array to be dot producted with the list of targets 
+            '''
+        target_desc_list = list(vars(target_desc).values())
+
+        target_desc = TargetDescription(
+                    np.eye(13)[target_desc_list[0]],
+                    np.eye(35)[target_desc_list[1]],
+                    np.eye(8)[target_desc_list[2]],
+                    np.eye(8)[target_desc_list[3]],
+                )
+        
+        for target_detail, target_probability in vars(target_desc).items():
```

Comment:
Using `vars` here is a bad idea because this code will break if we decide to add any more fields to `TargetDescription` and it breaks our type annotations for the distributions, so we can't be assured by the IDE that it's an ndarray.

I get that this is more elegant than writing out the same code for all 4 properties but in this case I think being explicit is better than being elegant.